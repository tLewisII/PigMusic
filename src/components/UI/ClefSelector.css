.clef-selector {
  text-align: center;
  padding: 2rem;
  background: #f5f5f5;
  border-radius: 12px;
  box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
  margin: 2rem auto;
  max-width: 600px;
}

.clef-selector h2 {
  color: #333;
  margin-bottom: 1.5rem;
  font-size: 2rem;
}

.clef-options {
  display: flex;
  justify-content: center;
  gap: 2rem;
  flex-wrap: wrap;
}

.clef-option {
  background: white;
  border: 3px solid #ddd;
  border-radius: 12px;
  padding: 2rem;
  cursor: pointer;
  transition: all 0.3s ease;
  min-width: 200px;
  position: relative;
  overflow: hidden;
}

.clef-option:hover {
  transform: translateY(-5px);
  box-shadow: 0 8px 16px rgba(0, 0, 0, 0.15);
  border-color: #007bff;
}

.clef-option:active {
  transform: translateY(-2px);
}

.clef-icon {
  font-size: 3rem;
  margin-bottom: 1rem;
}

.clef-option h3 {
  color: #333;
  margin-bottom: 0.5rem;
  font-size: 1.5rem;
}

.clef-option p {
  color: #666;
  font-size: 0.9rem;
  margin: 0;
}

.treble-clef:hover {
  border-color: #e91e63;
  background: linear-gradient(135deg, #fff 0%, #fce4ec 100%);
}

.bass-clef:hover {
  border-color: #2196f3;
  background: linear-gradient(135deg, #fff 0%, #e3f2fd 100%);
}

@media (max-width: 600px) {
  .clef-options {
    flex-direction: column;
    align-items: center;
  }
  
  .clef-option {
    width: 100%;
    max-width: 300px;
  }
}