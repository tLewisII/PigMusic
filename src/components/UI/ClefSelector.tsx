import React from 'react';
import { Clef } from '../../types/game';
import './ClefSelector.css';

interface ClefSelectorProps {
  onSelectClef: (clef: Clef) => void;
}

const ClefSelector: React.FC<ClefSelectorProps> = ({ onSelectClef }) => {
  return (
    <div className="clef-selector">
      <h2>Choose Your Clef</h2>
      <div className="clef-options">
        <button 
          className="clef-option treble-clef"
          onClick={() => onSelectClef('treble')}
          aria-label="Select Treble Clef"
        >
          <div className="clef-icon">ðŸŽ¼</div>
          <h3>Treble Clef</h3>
          <p>Higher notes (C4 - F5)</p>
        </button>
        
        <button 
          className="clef-option bass-clef"
          onClick={() => onSelectClef('bass')}
          aria-label="Select Bass Clef"
        >
          <div className="clef-icon">ðŸŽµ</div>
          <h3>Bass Clef</h3>
          <p>Lower notes (E2 - C4)</p>
        </button>
      </div>
    </div>
  );
};

export default ClefSelector;