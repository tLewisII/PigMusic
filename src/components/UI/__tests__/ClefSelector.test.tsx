import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import ClefSelector from '../ClefSelector';

describe('ClefSelector', () => {
  const mockOnSelectClef = jest.fn();

  beforeEach(() => {
    mockOnSelectClef.mockClear();
  });

  it('renders correctly', () => {
    const { container } = render(
      <ClefSelector onSelectClef={mockOnSelectClef} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('calls onSelectClef when treble button is clicked', () => {
    const { getByText } = render(
      <ClefSelector onSelectClef={mockOnSelectClef} />
    );
    
    fireEvent.click(getByText('Treble Clef'));
    expect(mockOnSelectClef).toHaveBeenCalledWith('treble');
  });

  it('calls onSelectClef when bass button is clicked', () => {
    const { getByText } = render(
      <ClefSelector onSelectClef={mockOnSelectClef} />
    );
    
    fireEvent.click(getByText('Bass Clef'));
    expect(mockOnSelectClef).toHaveBeenCalledWith('bass');
  });

  it('renders both clef options', () => {
    const { getByText } = render(
      <ClefSelector onSelectClef={mockOnSelectClef} />
    );
    
    expect(getByText('Treble Clef')).toBeInTheDocument();
    expect(getByText('Bass Clef')).toBeInTheDocument();
    expect(getByText('Higher notes (C4 - F5)')).toBeInTheDocument();
    expect(getByText('Lower notes (E2 - C4)')).toBeInTheDocument();
  });
});