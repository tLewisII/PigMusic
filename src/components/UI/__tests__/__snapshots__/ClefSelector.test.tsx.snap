// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClefSelector renders correctly 1`] = `
<div
  class="clef-selector"
>
  <h2>
    Choose Your Clef
  </h2>
  <div
    class="clef-options"
  >
    <button
      aria-label="Select Treble Clef"
      class="clef-option treble-clef"
    >
      <div
        class="clef-icon"
      >
        ðŸŽ¼
      </div>
      <h3>
        Treble Clef
      </h3>
      <p>
        Higher notes (C4 - F5)
      </p>
    </button>
    <button
      aria-label="Select Bass Clef"
      class="clef-option bass-clef"
    >
      <div
        class="clef-icon"
      >
        ðŸŽµ
      </div>
      <h3>
        Bass Clef
      </h3>
      <p>
        Lower notes (E2 - C4)
      </p>
    </button>
  </div>
</div>
`;
